hbase-server/src/test/java/org/apache/hadoop/hbase/io/hfile/bucket/TestBucketCache.java

```diff
@@ -258,17 +257,52 @@ public void testRetrieveFromFile() throws Exception {
     HBaseTestingUtility TEST_UTIL = new HBaseTestingUtility();
     Path testDir = TEST_UTIL.getDataTestDir();
     TEST_UTIL.getTestFileSystem().mkdirs(testDir);
-
     String ioEngineName = "file:" + testDir + "/bucket.cache";
+    testRetrievalUtils(testDir, ioEngineName);
+    int[] smallBucketSizes = new int[]{3 * 1024, 5 * 1024};
     String persistencePath = testDir + "/bucket.persistence";
+    BucketCache bucketCache = new BucketCache(ioEngineName, capacitySize, constructedBlockSize,
+            smallBucketSizes, writeThreads, writerQLen, persistencePath);
+    assertFalse(new File(persistencePath).exists());
+    assertEquals(0, bucketCache.getAllocator().getUsedSize());
+    assertEquals(0, bucketCache.backingMap.size());
+    TEST_UTIL.cleanupTestDir();
+  }
 
+  @Test
+  public void testRetrieveFromMMap() throws Exception {
+    HBaseTestingUtility TEST_UTIL = new HBaseTestingUtility();
```

Comment:
Do we get a new HBaseTestingUtility per test? If so, why not in the test setup  rather than at head of each test?
Also CAPITALIZATION of variables is for statics, not locals as here. 

Make a data member named testUtil and create the HTU instance in the @Before ?

Reply:
Done